radix-engine/src/system/system_modules/execution_trace/module.rs

```diff
@@ -91,10 +91,12 @@ impl From<&BucketSnapshot> for ResourceSpecifier {
 
 #[derive(Debug, Clone)]
 pub enum VaultOp {
-    Put(ResourceAddress, Decimal), // TODO: add non-fungible support
+    Put(ResourceAddress, Decimal),
     Take(ResourceAddress, Decimal),
+    // We intentionally disregard IDs and only use the amount for non-fungibles.
+    // This maintains backward compatibility for users of the `ResourceChange` struct.
+    TakeNonFungibles(ResourceAddress, Decimal),
     TakeAdvanced(ResourceAddress, Decimal),
-    Recall(ResourceAddress, Decimal),
```

Comment:
Is it worth mentioning that Recall is not supported (and why)?
And perhaps (somewhere) that this feature isn't comprehensive/a good abstraction, and will be deprecated in the node's Core API at Dugong, with the intention to have it removed at a later point?